### DISCUSSION TITLE
modbus wired

### DISCUSSION THREAD

**Discussion by sp2003** [alexdelprete/ha-abb-powerone-pvi-sunspec#115] [General]
Ciao Alex

per caso hai in progetto di implementare la tua integrazione con il supporto al modbus filare?
Quella nativa in HA visualizza solo 4 entità : Status , Energia Totale, Potenza in uscita e Temperatura mentre la tua è molto più completa.
Il mio inverter è un
Aurora 6kW outdoor - N/A - Transformerless Version - PV Version (-3G03-)
Firmware: C.0.3.4

In precedenza leggevo e immagazinavo i dati con 123solar e avevo molti più dati

**Comment by alexdelprete** [comment]
Ciao,

cos'è il modbus "filare"? intendi la porta seriale?

**Reply by sp2003** [reply]
si si, con collegamento via cavo tra inverter e pc

**Reply by alexdelprete** [reply]
ma comprati una VSN300 no? :)

oppure per 20-30 eur un gateway rs485-ethernet/wifi.

**Reply by sp2003** [reply]
ho già tutto steso, collegato e funzionante. qualche mese fa provai questo ma non andò bene 
![17074863471662877824701065418882](https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/assets/34820831/47d8bd1c-3649-4eb5-903e-5d74a7fc38f9)

**Comment by alexdelprete** [comment]
Io comprai questo, prima di ordinare la VSN300 e funzionava molto bene. 

![image](https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/assets/7027842/1acfa8fc-9645-46c8-8ae3-8ad46d7693c9)

**Comment by sp2003** [comment]
sicuramente, via modbus TCP, ne esistono tanti adattatori ma io volevo rimanere nella condizione hardware attuale ecco perché ti chiedevo se avessi in programma un aggiornamento della tua integrazione custom

**Comment by alexdelprete** [comment]
Ci penso e verifico quanto debba modificare...se non è tanta roba la implementerò.

**Reply by sp2003** [reply]
grazie!

**Comment by sp2003** [comment]
Ciao Alex, hai avuto modo di vedere se la cosa è fattibile?

**Reply by alexdelprete** [reply]
ci abbiamo provato con una persona che ha il trio-20 ed usa un gateway Elfin.

Purtroppo riusciamo a comunicare con l'inverter via seriale, ma l'inverter usa la mappatura dei registri Modbus RTU, e non quella Sunspec. Ora sta contattando FIMER per verificare se si possa abilitare qualcosa per fargli traslare gli indirizzi in standard Sunspec. Alcuni inverter lo fanno altri no.

Quindi non credo che lo implementerò. Se fosse stato solo la gestione della seriale sarebbe stato fattibile.

Mi spiace.

**Reply by sp2003** [reply]
cavolo peccato!

ho utilizzato per anni questo software []https://www.curtronics.com/Solar/AuroraData.html e riuscivo tranquillamente a leggere tutti i dati come con il tuo.
magari se hai tempo e vuoi dargli un occhiata.
Qualche mese fa contattai l'autore via mail perché avevo problemi nella compilazioen del suo software e fu gentilissimo.

**Reply by alexdelprete** [reply]
Lo conosco da anni quel software. Ma poi 3-4 anni fa installai HA e decisi di passare a qualcosa di più standard (Sunspec) con la VSN300. Dovresti fare la stessa cosa.

**Comment by sbetti1973** [comment]
ciao Alex non rieco a collegare il mio poweone in HA , aurora conect funziona , ho una schedina modbus/lan questo è il log di errore Registratore: aiohttp.server
Fonte: /usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py:421
Prima occorrenza: 14:07:43 (1 occorrenze)
Ultima registrazione: 14:07:43

Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_app.py", line 543, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/web_middlewares.py", line 114, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 91, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 100, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 25, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 79, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 235, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 31, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/http.py", line 68, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 72, in with_admin
    return await func(self, request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 197, in post
    return await super().post(request, flow_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 72, in wrapper
    result = await method(view, request, data, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 111, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 359, in async_configure
    result = await self._async_configure(flow_id, user_input)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 405, in _async_configure
    result = await self._async_handle_step(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 502, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/abb_powerone_pvi_sunspec/config_flow.py", line 112, in async_step_user
    uid = await self.test_connection(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/abb_powerone_pvi_sunspec/config_flow.py", line 85, in test_connection
    self.api_data = await self.api.async_get_data()
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/abb_powerone_pvi_sunspec/api.py", line 190, in async_get_data
    if self.connect():
       ^^^^^^^^^^^^^^
  File "/config/custom_components/abb_powerone_pvi_sunspec/api.py", line 167, in connect
    raise ConnectionError(f"Inverter not active on {self._host}:{self._port}")
custom_components.abb_powerone_pvi_sunspec.api.ConnectionError: Inverter not active on 192.168.2.157:502

**Reply by alexdelprete** [reply]
come è collegato l'inverter? VSN300? il messaggio è chiaro: non risponde all'indirizzo 192.168.2.157 sulla porta 502

**Reply by alexdelprete** [reply]
se usi un bridge modbus, l'inverter supporta solo il protocollo Aurora. La mia integrazione si basa su Modbus TCP Sunspec, quindi serve che l'inverter supporti la mappatura dei registri Sunspec. La scheda VSN300 e la VSN700 fanno ciò.

La cosa più semplice è verificare se il tuo inverter supporti la VSN300 e acquistarla.

**Comment by sbetti1973** [comment]
purtroppo non la supporta , grazie del supporto e buon we

Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by alexdelprete** [reply]
mi sembra strano. che modello di inverter hai?

**Comment by sbetti1973** [comment]
dovrebbe essere questo e al momento mi collego con un waveshare collegato alla lan

Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by alexdelprete** [reply]
questo quale? non vedo nulla...

**Comment by sbetti1973** [comment]
è collegato con un convertitore

Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by sbetti1973** [comment]
in allegato

Inviato da Outlook per Android<https://aka.ms/AAb9ysg>
________________________________

**Reply by alexdelprete** [reply]
non c'è nessun allegato...usa github per scrivere, non la posta.

**Comment by sbetti1973** [comment]
eccolo
![Screenshot_20240309_154010_Samsung Internet](https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/assets/162138699/988932d3-aeb3-405c-b6c0-2c6e61d6fca6)
![Screenshot_20240309_154010_Samsung Internet](https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/assets/162138699/631d1704-ced6-4169-8bbf-6109f4c909c4)

**Reply by alexdelprete** [reply]
questo è il bridge waveshare, ho chiesto il modello di inverter :)

**Comment by sbetti1973** [comment]
è il pvi-3.0-outd

**Reply by alexdelprete** [reply]
e chi ti ha detto che non supporta la VSN300?

Il tuo inverter è compatibile sia con la VSN300 che la VSN700:

![image](https://github.com/alexdelprete/ha-abb-powerone-pvi-sunspec/assets/7027842/6bbf44e1-e884-4b9c-a944-dd4ff33be0fd)

**Reply by sbetti1973** [reply]
è una buona notizia se fosse dove la trovo ,? e a che prezzo

**Reply by alexdelprete** [reply]
dovrebbe costare intorno ai 150. devi cercarla online, io l'ho comprata 3 anni fa.
